src/pages/iou/IOUModal.js

```diff
@@ -0,0 +1,173 @@
+
+/**
+ * IOU modal for requesting money and splitting bills.
+ */
+const propTypes = {
+
+    /* Onyx Props */
+    // Url currently in view
+    currentURL: PropTypes.string,
+
+    // IOU modal data
+    iouData: PropTypes.shape({
+        isLoadingCurrency: PropTypes.bool.isRequired,
+        isLoadingParticipants: PropTypes.bool.isRequired,
+        createReportInProgress: PropTypes.bool.isRequired,
+    }),
+};
+
+const StepType = {
+    IOUAmount: 'IOUAmount',
+    IOUParticipants: 'IOUParticipants',
+    IOUConfirm: 'IOUConfirm',
+};
+
+const defaultProps = {
+    currentURL: '',
+    iouData: {
+        isLoadingCurrency: true,
+        isLoadingParticipants: true,
+        createReportInProgress: false,
+    },
+};
+
+class IOUModal extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            steps: [StepType.IOUAmount, StepType.IOUParticipants, StepType.IOUConfirm],
+            currentStepIndex: 0,
+            hasMultipleParticipants: this.props.currentURL === ROUTES.IOU_BILL_SPLIT,
+        };
+
+        this.getTitleForStep = this.getTitleForStep.bind(this);
+        this.navigateToPreviousStep = this.navigateToPreviousStep.bind(this);
+        this.navigateToNextStep = this.navigateToNextStep.bind(this);
+    }
+
+    componentDidMount() {
+        getPreferredCurrency();
+    }
+
+    /**
+     * Returns the title for the currently selected page
+     *
+     * @return {String}
+     */
+    getTitleForStep() {
+        switch (this.state.steps[this.state.currentStepIndex]) {
+            case StepType.IOUAmount:
+                return 'Amount';
+            case StepType.IOUParticipants:
+                return 'Participants';
+            case StepType.IOUConfirm:
+                return 'Confirm';
+            default:
+                return '';
+        }
+    }
+
+    /**
+     * Navigate to the next IOU step
+     */
+    navigateToPreviousStep() {
+        this.setState(prevState => ({currentStepIndex: prevState.currentStepIndex - 1}));
+    }
+
+    /**
+     * Navigate to the previous IOU step
+     */
+    navigateToNextStep() {
+        this.setState(prevState => ({currentStepIndex: prevState.currentStepIndex + 1}));
+    }
+
+    render() {
+        return (
+            <>
+                <View style={[styles.headerBar, true && styles.borderBottom]}>
+                    <View style={[
+                        styles.dFlex,
+                        styles.flexRow,
+                        styles.alignItemsCenter,
+                        styles.flexGrow1,
+                        styles.justifyContentBetween,
+                        styles.overflowHidden,
+                    ]}
+                    >
+                        {this.state.currentStepIndex > 0
+                        && (
+                            <TouchableOpacity
+                                onPress={() => this.navigateToPreviousStep()}
```

Comment:
```suggestion
                                onPress={this.navigateToPreviousStep}
```
(then bind it in the constructor)